Sources/JobsPostgres/PostgresJobsQueue.swift

```diff
@@ -160,9 +160,10 @@ public final class PostgresJobQueue: JobQueueDriver {
 
     func popFirst() async throws -> QueuedJob<JobID>? {
         do {
-            return try await self.client.withConnection { connection -> QueuedJob? in
+            return try await self.client.withTransaction(logger: self.logger) { connection -> QueuedJob? in
```

Comment:
There's one issue with this. At the bottom of this `withTransaction` block if the job fails to decode, it throws a `JobQueueError.decodeJobFailed` error which will then cancel the whole of the transaction. Thus the job is never popped off the queue and flagged as failed and the next time it comes to pop a job off the queue it'll do exactly the same thing with the same job because it is still at the top of the queue.

We could return a `Result<QueuedJob<JobID>, Error>` from the `withTransaction` block. And if it is an error throw the error once outside of the `withTransaction` block

Reply:
Understood! Eventually, I wanted to move the table to [this schema](https://github.com/hummingbird-project/hummingbird-postgres/issues/19) which would afford us to drop jobs that has failed for N number of times. I will update the PR with the above suggestion.